x-pack/plugins/infra/server/lib/domains/log_entries_domain/convert_document_source_to_log_item_fields.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isArray, isPlainObject } from 'lodash';
+import { JsonObject } from 'x-pack/plugins/infra/common/typed_json';
+import { InfraLogItemField } from '../../../graphql/types';
+
+const isJsonObject = (subject: any): subject is JsonObject => {
+  return isPlainObject(subject);
+};
+
+const serializeValue = (value: any): string => {
+  if (isArray(value) || isPlainObject(value)) {
+    return JSON.stringify(value);
+  }
+  return `${value}`;
+};
+
+export const convertDocumentSourceToLogItemFields = (
```

Comment:
trigger warning :wink:

Reply:
huh?

Reply:
looks very `flatten`ish to me

Reply:
I assume you mean `_.flatten` from lodash? The input on that is an array where this is an object that is converted into an array of objects with a path key and a value. We could import `src/utils/get_flattened_object.test.ts` from Kibana then convert the output to the array we need. I'm inclined to stick with this solution because it we control it and it's only 8 lines of code.

Reply:
sorry for causing confusion here. it was intended as a funny reference to the many occasions where we witnessed a variation of this being written :see_no_evil: